model/riscv_insts_vext_utils.sail

```diff
@@ -0,0 +1,1145 @@
+  write_fflags(fflags);
+  result_val
+}
+
+val fp_mul : forall 'm, 'm in {16, 32, 64}. (bits(3), bits('m), bits('m)) -> bits('m) effect {escape, rreg, undef, wreg}
+function fp_mul(rm_3b, op1, op2) = {
+  let (fflags, result_val) : (bits_fflags, bits('m)) = match 'm {
+    16  => riscv_f16Mul(rm_3b, op1, op2),
+    32  => riscv_f32Mul(rm_3b, op1, op2),
+    64  => riscv_f64Mul(rm_3b, op1, op2)
+  };
+  write_fflags(fflags);
+  result_val
+}
+
+val fp_div : forall 'm, 'm in {16, 32, 64}. (bits(3), bits('m), bits('m)) -> bits('m) effect {escape, rreg, undef, wreg}
+function fp_div(rm_3b, op1, op2) = {
+  let (fflags, result_val) : (bits_fflags, bits('m)) = match 'm {
+    16  => riscv_f16Div(rm_3b, op1, op2),
+    32  => riscv_f32Div(rm_3b, op1, op2),
+    64  => riscv_f64Div(rm_3b, op1, op2)
+  };
+  write_fflags(fflags);
+  result_val
+}
+
+val fp_muladd : forall 'm, 'm in {16, 32, 64}. (bits(3), bits('m), bits('m), bits('m)) -> bits('m) effect {escape, rreg, undef, wreg}
+function fp_muladd(rm_3b, op1, op2, opadd) = {
+  let (fflags, result_val) : (bits_fflags, bits('m)) = match 'm {
+    16  => riscv_f16MulAdd(rm_3b, op1, op2, opadd),
+    32  => riscv_f32MulAdd(rm_3b, op1, op2, opadd),
+    64  => riscv_f64MulAdd(rm_3b, op1, op2, opadd)
+  };
+  write_fflags(fflags);
+  result_val
+}
+
+val fp_nmuladd : forall 'm, 'm in {16, 32, 64}. (bits(3), bits('m), bits('m), bits('m)) -> bits('m) effect {escape, rreg, undef, wreg}
+function fp_nmuladd(rm_3b, op1, op2, opadd) = {
+  let op1 = negate_fp(op1);
+  let (fflags, result_val) : (bits_fflags, bits('m)) = match 'm {
+    16  => riscv_f16MulAdd(rm_3b, op1, op2, opadd),
+    32  => riscv_f32MulAdd(rm_3b, op1, op2, opadd),
+    64  => riscv_f64MulAdd(rm_3b, op1, op2, opadd)
+  };
+  write_fflags(fflags);
+  result_val
+}
+
+val fp_mulsub : forall 'm, 'm in {16, 32, 64}. (bits(3), bits('m), bits('m), bits('m)) -> bits('m) effect {escape, rreg, undef, wreg}
+function fp_mulsub(rm_3b, op1, op2, opsub) = {
+  let opsub = negate_fp(opsub);
+  let (fflags, result_val) : (bits_fflags, bits('m)) = match 'm {
+    16  => riscv_f16MulAdd(rm_3b, op1, op2, opsub),
+    32  => riscv_f32MulAdd(rm_3b, op1, op2, opsub),
+    64  => riscv_f64MulAdd(rm_3b, op1, op2, opsub)
+  };
+  write_fflags(fflags);
+  result_val
+}
+
+val fp_nmulsub : forall 'm, 'm in {16, 32, 64}. (bits(3), bits('m), bits('m), bits('m)) -> bits('m) effect {escape, rreg, undef, wreg}
+function fp_nmulsub(rm_3b, op1, op2, opsub) = {
+  let opsub = negate_fp(opsub);
+  let op1 = negate_fp(op1);
+  let (fflags, result_val) : (bits_fflags, bits('m)) = match 'm {
+    16  => riscv_f16MulAdd(rm_3b, op1, op2, opsub),
+    32  => riscv_f32MulAdd(rm_3b, op1, op2, opsub),
+    64  => riscv_f64MulAdd(rm_3b, op1, op2, opsub)
+  };
+  write_fflags(fflags);
+  result_val
+}
+
+val fp_class : forall 'm, 'm in {16, 32, 64}. bits('m) -> bits('m)
+function fp_class(xf) = {
+  let result_val_10b : bits(10) =
+    if      f_is_neg_inf     (xf) then 0b_00_0000_0001
```

Comment:
Nitpick: I think the formatting here is a bit unorthodox. We could keep the `then` aligned, but I would remove the space between the function `f_is_neg_inf` etc and `(xf)`

Reply:
I'll modify this part in the next fixup submission.